### PR TITLE
Added logic for toggling minable robots

### PR DESCRIPTION
**Pull request by SimonFlapse**
I couldn't find anything in the upcoming feature section of the wiki that robots shouldn't be minable in 0.17

This adds a check everything a player selects a construction robot that:
- Checks is the construction robot is owned by a specific player
  - If not: Sets the construction robot to minable = false
  - If: Sets the construction robot to minable = true

This will practically prevent other players from intentional, or unintentional, mine a players personal construction robots.

You might be able to mine a robot if you select it first and the owner selects it afterwards, and you do not remove your selection. I think this is an edge case that's not worth trying to fix.

### CODE REVIEW

File: features/save_bots.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
+local Event = require 'utils.event'
+local Game = require 'utils.game'
+
+local function preserve_bot(event)
+    local player = Game.get_player_by_index(event.player_index)
```
Comment (Valansch) [inline features/save_bots.lua]: Move this line to line 19 please. Preserve_bot will be called alot.

**Review by Valansch — DISMISSED** [review]
Great idea!
One minor thing.

File: features/save_bots.lua (hunk 1 above)
Comment (SimonFlapse) [reply features/save_bots.lua]: Yeah I know, but I can't :( The event only passes on the previous selected item, not the new one. So I need the player

Comment (Valansch) [reply features/save_bots.lua]: Oh i missed line 6. Thats a bummer.

File: features/save_bots.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    local entity = player.selected
+
+    if entity == nil or not entity.valid then
+        return
+    end
+
+    if entity.name == 'construction-robot' then
+        local logistic_cell = entity.logistic_network.cells[1]
```
Comment (grilledham) [inline features/save_bots.lua]: Can you use <https://lua-api.factorio.com/latest/LuaEntity.html#LuaEntity.logistic_cell> instead?
Also do we need to check if cell is nil, what happens if you select a robot and there are no logistic_cell on the map?

Comment (SimonFlapse) [reply features/save_bots.lua]: I tried initially but didn't seem to work very well. And the player will always only have one cell. My local changes have a check for a nil cell

Comment (SimonFlapse) [reply features/save_bots.lua]: image

**Comment by mheguy** [comment]
Because of the small size and scope, this would be a good fit for `redmew_qol`.

**Comment by SimonFlapse** [comment]
Is it a problem to have it isolated? I could understand if you wanted to reuse the same event to reduce overhead.

**Comment by mheguy** [comment]
Not a problem to have it isolated, nope. Was just a suggestion since it's a good fit.

**Comment by SimonFlapse** [comment]
Updated to 0.17 and moved to Redmew-QoL.lua

Tested on S9.

**Review by mheguy — CHANGES_REQUESTED** [review]
You have two copies of the module in this PR.

File: features/redmew_qol.lua [hunk 3]
```diff
@@ -139,6 +139,45 @@ local loader_check_token =
     end
 )
 
+--- Sets construction robots that are not part of a roboport to unminabe
+-- if the player selecting them are not the owner of them.
+local function preserve_bot(event)
+    local player = Game.get_player_by_index(event.player_index)
+    local entity = player.selected
+
+    if entity == nil or not entity.valid then
```
Comment (plague006) [inline features/redmew_qol.lua]: As a style choice, I'm wondering why you use `== nil` and `not` both?

**Review by mheguy — APPROVED** [review]

**Pull request merged**